### DISCUSSION TITLE
[@nx/angular/mf] setRemoteDefinition(): A stateful way to add new remotes dinamically

### DISCUSSION THREAD

**Discussion by menosprezzi** [nrwl/nx#27050] [Feature Requests]
Hello Nrwl team and Nx community!
At my current company, we are using angular/mf from some years, and fully replaced angular-architects package to the best devx of @nx/angular/mf!

We are building a set of auto-configurables MicroFrontEnds based on Libs.
In these libs we expose the main contract of the MFE (preventing breaks in our consumers), share declarations/injection tokens and we do an auto-configuration, setting up the MFE remote based on the environment and the host Angular version, because we need to maintain multiple MFE versions published so we can do a smooth rollout between the many consumers.

The problem is that `@nx/angular/mf`s `setRemoteDefinitions` method overwrite the previous state (as expected), so if a consumer uses more than 1 of this solutions, the state is overwritten and the last MFEs wont load properly anymore.

Therefore, I'd like to propose a new API called `addRemoteDefinition(remoteName: string, remoteUrl: string)` (or `setRemoteDefinition`) that adds or replace a new remote definition in the remotes map, without overwrite it.

If you guys agree, I'll be able and happy to contribute with a PR!

Thanks!

**Comment by saratiedt** [comment]
I have the same problem.

**Comment by menosprezzi** [comment]
I've already left a PR open to address that.
#27051

**Comment by menosprezzi** [comment]
Awesome! The PR was approved! Closing this discussion by now!
